conftest.py

```diff
@@ -29,15 +29,25 @@ def pytest_addoption(parser):
 
 
 def pytest_collection_modifyitems(config, items):
-
-    # FeatureHasher is not compatible with PyPy
-    if platform.python_implementation() == 'PyPy':
-        skip_marker = pytest.mark.skip(
-            reason='FeatureHasher is not compatible with PyPy')
-        for item in items:
-            if item.name.endswith(('_hash.FeatureHasher',
-                                   'text.HashingVectorizer')):
-                item.add_marker(skip_marker)
+    for item in items:
+        # FeatureHasher is not compatible with PyPy
+        if (item.name.endswith(('_hash.FeatureHasher',
+                                'text.HashingVectorizer'))
+                and platform.python_implementation() == 'PyPy'):
+            marker = pytest.mark.skip(
+                reason='FeatureHasher is not compatible with PyPy')
+            item.add_marker(marker)
+        # Known failure on with GradientBoostingClassifier on ARM64
+        elif (item.name.endswith('GradientBoostingClassifier')
+                and platform.machine() == 'aarch64'):
```

Comment:
I'm still confused on the "aarch64" vs "ARM64" since the latter can also happen apparently https://github.com/python/cpython/blob/0c4f0f3b29d84063700217dcf90ad6860ed71c70/Lib/test/test_regrtest.py#L662

Anyway the the issue where this doctest failure was originally reported https://github.com/scikit-learn/scikit-learn/issues/17797 was also aarch64 so it's probably fine to merge as is